### PR TITLE
*(dm): add some warning and adjust some config

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5220

### What is changed and how it works?

No. 1, 2, 5, 6 subtask of #5220 

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported variable/fields change

Side effects

Related changes

### Release note 

```release-note
Some commands and configuration will be deprecated in future:
- `start-relay` / `stop-relay` with worker name
- `is-sharding` in task configuration
```

### CODE REVIEW

File: dm/dm/ctl/master/check_task.go [hunk 1]
```diff
@@ -63,10 +65,26 @@ func checkTaskFunc(cmd *cobra.Command, _ []string) error {
...
+	for i := range lines {
+		if bytes.HasPrefix(lines[i], []byte("is-sharding")) {
+			isShardingSet = true
+			break
+		}
+	}
+	task := config.NewTaskConfig()
+	yamlErr := task.RawDecode(string(content))
+	// if can't parse yaml, we ignore this check
+	if yamlErr == nil && isShardingSet && !task.IsSharding && task.ShardMode != "" {
```
Comment (Ehco1996) [inline dm/dm/ctl/master/check_task.go:80]: what if `yamlErr != nil` , we can ignore this?

File: dm/dm/ctl/master/operate_source.go [hunk 2]
```diff
@@ -136,6 +136,9 @@ func operateSourceFunc(cmd *cobra.Command, _ []string) error {
 			}
 			content = []byte(yamlStr)
 		}
+		if cfg.RelayDir != "" {
+			common.PrintLinesf("`relay-dir` in source config will be deprecated soon, please use `relay-dir` in worker config instead")
```
Comment (Ehco1996) [inline dm/dm/ctl/master/operate_source.go:140]: do we also add this warnning to openapi ?

File: dm/tests/dmctl_basic/check_list/check_task.sh [hunk 3]
```diff
@@ -27,6 +27,14 @@ function check_task_not_pass() {
 		"\"result\": false" 1
 }
 
+function check_task_not_pass_with_message() {
+	task_conf=$1
+	error_message=$2
```
Comment (Ehco1996) [inline dm/tests/dmctl_basic/check_list/check_task.sh:32]: var  `error_message` is not used

**Review by Ehco1996 — COMMENTED** [review]
restLGTM

File: dm/dm/ctl/master/start_task.go [hunk 4]
```diff
@@ -67,6 +68,21 @@ func startTaskFunc(cmd *cobra.Command, _ []string) error {
...
+		if bytes.HasPrefix(lines[i], []byte("is-sharding")) {
+			isShardingSet = true
+			break
+		}
+	}
+	if isShardingSet && !task.IsSharding && task.ShardMode != "" {
+		common.PrintLinesf("The behaviour of `is-sharding` and `shard-mode` is conflicting. `is-sharding` is deprecated, please use `shard-mode` only.")
+		return errors.New("please check output to see error")
+	}
+
```
Comment (GMHDBJD) [inline dm/dm/ctl/master/start_task.go:85]: Maybe extract to a function in common package

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: dm/dm/ctl/master/check_task.go (hunk 1 above)
Comment (lance6716) [reply dm/dm/ctl/master/check_task.go:80]: keep old behaviour

File: dm/dm/ctl/master/operate_source.go (hunk 2 above)
Comment (lance6716) [reply dm/dm/ctl/master/operate_source.go:140]: cc @sunzhaoyang

File: dm/dm/ctl/master/start_task.go (hunk 4 above)
Comment (lance6716) [reply dm/dm/ctl/master/start_task.go:85]: will leave for future intern :smile:

File: dm/dm/ctl/master/operate_source.go (hunk 2 above)
Comment (sunzhaoyang) [reply dm/dm/ctl/master/operate_source.go:140]: No need. We should describe this information in the openapi's documents .

**Review by Ehco1996 — APPROVED** [review]
LGTM

**Pull request merged**